test/integration/compile-test.coffee

```diff
@@ -91,33 +79,22 @@ describe('compile() · all API description formats', ->
         )
       )
 
-      it('is compiled into zero transactions', ->
-        assert.deepEqual(compilationResult.transactions, [])
+      it('produces some annotations and no transactions', ->
```

Comment:
`produces some annotations...` seems a little bit too loose to me, is there any good reason for this? Ditto for other occurrences.

Reply:
It is because of differences in formats. E.g. APIB produces no warning, but Swagger produces one. Every time I've used this, I added further assertions in subsequent `it`s, inspecting the real numbers and types of annotations in the result. At least I hope I did not forget it anywhere.

I was thinking whether this could be done in more declarative way as well, because writing all the `assert.atMost` and `annotations.filter()` etc. is annoying and error-prone. So having sth like

```
{
  annotations:
    {type: 'warning', component: 'apiDescriptionParser', count: 1}
    {type: 'error', countAtMost: 2}
  transactions: 2
}
```

I don't know yet. Just thoughts.

Reply:
🤔 I see. I like the solution you proposed (`countAtMost` property, or similar) 👍